c2cciutils/checks.py

```diff
@@ -0,0 +1,654 @@
+        if not os.path.exists(filename):
+            error(
+                "required_workflows",
+                "The workflow '{}' is required".format(filename),
+                filename,
+            )
+            result = False
+            continue
+
+        if isinstance(config, dict):
+            continue
+
+        with open(filename) as open_file:
+            workflow = yaml.load(open_file, yaml.SafeLoader)
+
+        for name, job in workflow.get("jobs").items():
+            if "if" in conf:
+                if job.get("if") != conf["if"]:
+                    error(
+                        "required_workflows",
+                        "The workflow '{}', job '{}' does not have the following if '{}'".format(
+                            filename, name, conf["if"]
+                        ),
+                        filename,
+                    )
+                    result = False
+            if conf.get("noif", False):
+                if "if" in job:
+                    error(
+                        "required_workflows",
+                        "The workflow '{}', job '{}' should not have a if".format(filename, name),
+                        filename,
+                    )
+                    result = False
+            if "strategy-fail-fast" in conf:
+                if job.get("strategy", {}).get("fail-fast") != conf["strategy-fail-fast"]:
+                    error(
+                        "required_workflows",
+                        "The workflow '{}', job '{}' does not have the strategy/fail-fast as {}".format(
+                            filename, name, conf["strategy-fail-fast"]
+                        ),
+                        filename,
+                    )
+                    result = False
+            if "runs_re" in conf:
+                for run in conf["runs_re"]:
+                    corresponding_steps = [
+                        step for step in job["steps"] if re.match(step.get("run"), run) is not None
+                    ]
+                    if len(corresponding_steps) == 0:
+                        error(
+                            "required_workflows",
+                            "The workflow '{}', job '{}' doesn't have the step that runs '{}'.".format(
+                                filename, name, run
+                            ),
+                            filename,
+                        )
+                        result = False
+    return result
+
+
+def versions(config, full_config, _):
+    """
+    Check that all the versions corresponds (the base is the version from SECURITY.md).
+    The columns `Version` and `Supported Until`should be present.
+    The `Supported Until` should contains dates formatted as `dd/mm/yyyy`, or `Unsupported`
+    (we ignore thoes lines), or `Best effort`.
+
+    config is like:
+        extra_versions: # Versions that not in the SECURITY.md and should be consider as it.
```

Comment:
```suggestion
        extra_versions: # Versions that are not in the SECURITY.md but should still be consided.
```
But still not understand considered for what ?